### DISCUSSION TITLE
@prisma/client did not initialize yet

### DISCUSSION THREAD

**Discussion by abergwall** [prisma/prisma#19208] [Q&A]
### Bug description

I've tried everything and read every thread on the internet but I still keep getting this error.
It works on local but not in prod.

I've built a Nuxt3 app using trpc, prisma, supabase and try to deploy on Vercel.
I only call prisma from the server folder and I am not sure if I need to create some middleware or server plugin function to get prisma to initialize?

My package JSON looks like this : 
```

{
  "name": "nuxt-app",
  "private": true,
  "scripts": {
    "build": "prisma generate && prisma migrate deploy && nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "devDependencies": {
    "@nuxtjs/tailwindcss": "^6.6.6",
    "@types/node": "^18",
    "@vueuse/core": "^10.1.2",
    "@vueuse/nuxt": "^10.1.2",
    "autoprefixer": "^10.4.14",
    "nuxt": "^3.4.2",
    "postcss": "^8.4.23",
    "prisma": "^4.14.0",
    "tailwindcss": "^3.3.1"
  },
  "dependencies": {
    "@headlessui/vue": "^1.7.13",
    "@nuxt/types": "^2.16.3",
    "@trpc/client": "^10.25.0",
    "@trpc/server": "^10.25.0",
    "@prisma/client": "^4.11.0",
    "happy-dom": "^9.9.2",
    "trpc-nuxt": "^0.9.0",
    "zod": "^3.21.4"
  }
}

```

### How to reproduce

NA

### Prisma information

prisma/schema.prisma
```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
  output   = "./generated/client"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

Using Prisma in server/trpc/routers/documents.ts

```ts
import { publicProcedure, router } from "../trpc";
import { PrismaClient } from "@prisma/client";
import { z } from "zod";

const prisma = new PrismaClient({});

export const documentRouter = router({
  getOne: publicProcedure
    .input(z.object({ id: z.number() }))
    .query(async ({ ctx, input }) => {
      const document = await prisma.dokument.findUnique({
        where: { hangar_id: input.id },
        include: { debatt: true },
      });
      return document;
    }),

  getAll: publicProcedure.query(async ({ ctx }) => {
    const documents = await prisma.dokument.findMany({
      orderBy: { datum: "desc" },
      include: { debatt: true },
      take: 20,
    });

    return documents;
  }),
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by ludralph** [comment]
Hi @abergwall 👋 

Can you add the `prisma generate` command to the postinstall script as mentioned in this [guide](https://www.prisma.io/docs/guides/other/troubleshooting-orm/help-articles/vercel-caching-issue#a-custom-postinstall-script).

**Reply by abergwall** [reply]
@ludralph I've tried that but it doesn't make any difference

**Reply by janpio** [reply]
Can you create a small, minimal reproduction of the problem with a new app and share that with us? It would be optimal if we could just try to deploy this ourselves to _see_ what is happening for you.

**Reply by ludralph** [reply]
Hi @abergwall, Did you get to see Jan's message? A reproduction will be useful in this regard.

**Comment by david-ying-chan** [comment]
Have you tried 'import { PrismaClient } from ".prisma/client";' in code?

After I updated Primsa form 4.13.0 to 4.16.2, "@prisma/client" no longer works for me, yet ".prisma/client" works.

**Reply by david-ying-chan** [reply]
It's funny, actually when I upgrade prisma version from 4.13.0 to 4.16.2, I switch to pnpm instead of npm too. So to figure out the exact root cause, I rollback to npm with 4.16.2, and "@prisma/client" works. Yet surprisingly, I then change to pnpm with 4.16.2, it still works. So maybe removing node_modules and lockfile and install again will help?

**Reply by david-ying-chan** [reply]
One final update: both npm and pnpm work in my local machine, yet when deployed to a jenkins server, pnpm will fail(with no caches). So I believe this is still **a pnpm related issue** somehow.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
